apps/docs/content/docs/framework/events/index.mdx

```diff
@@ -1,156 +1,678 @@
 
-- **User messages** - Direct input from end users
-- **Agent replies** - Responses and actions from agents
-- **Tool interactions** - Function calls and their results
-- **State changes** - Updates to session state and artifacts
-- **Control signals** - Flow control and error conditions
+Technically, it's an instance of the `Event` class, which builds upon the basic `LlmResponse` structure by adding essential ADK-specific metadata and an `actions` payload.
+
+```typescript path=null start=null
+// Conceptual Structure of an Event (TypeScript)
+import { Event, EventActions } from '@iqai/adk';
+
+class Event extends LlmResponse { // Simplified view
+  // --- LlmResponse fields ---
+  content?: any;                 // Event payload (text, function calls, etc.)
+  partial?: boolean;             // True for streaming chunks
+  // ... other response fields ...
+
+  // --- ADK specific additions ---
+  author: string;                // 'user' or agent name
+  invocationId: string;          // ID for the whole interaction run
+  id: string;                    // Unique ID for this specific event
+  timestamp: number;             // Creation time (milliseconds since epoch)
+  actions: EventActions;         // Important for side-effects & control
+  branch?: string;               // Hierarchy path
+  // ...
+}
+```
+
+Events are central to ADK's operation for several key reasons:
+
+1. **Communication:** They serve as the standard message format between the user interface, the `Runner`, agents, the LLM, and tools. Everything flows as an `Event`.
+
+2. **Signaling State & Artifact Changes:** Events carry instructions for state modifications and track artifact updates. The `SessionService` uses these signals to ensure persistence. Changes are signaled via `event.actions.stateDelta` and `event.actions.artifactDelta`.
+
+3. **Control Flow:** Specific fields like `event.actions.transferToAgent` or `event.actions.escalate` act as signals that direct the framework, determining which agent runs next or if a loop should terminate.
+
+4. **History & Observability:** The sequence of events recorded in `session.events` provides a complete, chronological history of an interaction, invaluable for debugging, auditing, and understanding agent behavior step-by-step.
+
+In essence, the entire process, from a user's query to the agent's final answer, is orchestrated through the generation, interpretation, and processing of `Event` objects.
+
+## Understanding and Using Events
+
+As a developer, you'll primarily interact with the stream of events yielded by the `Runner`. Here's how to understand and extract information from them:
 
 <Callout type="info">
-Events serve as the standard message format between the user interface, the `Runner`, agents, the LLM, and tools. Everything flows as an `Event`.
+The specific parameters or method names may vary slightly by implementation. Refer to the TypeScript API documentation for exact method signatures.
```

Comment:
Remove this callout, it is not necessary as our adk only has one language. Google's adk has Java and Python, that's why they added the callout to their doc. Take note of this when you are updating other docs.